### PR TITLE
Move to use new sanitizer passes

### PR DESCRIPTION
**Pull request by thewilsonator**
see #3961

I chose to enable them from LLVM13+ partly because that's when the old ones were deprecated and partly to ensure that the CI tests them. That can be easily changed if desired.

Still to go is replacement for `createPGOIndirectCallPromotionLegacyPass`.

### CODE REVIEW

**Comment by kinke** [comment]
The sanitizers are probably just a minor aspect of switching to the 'new' pass manager (https://llvm.org/docs/NewPassManager.html). We also need to port *our* passes (https://forum.dlang.org/thread/mqcmbyvhdeoitcipyzkj@forum.dlang.org) before we can use it.

File: gen/optimizer.cpp [hunk 1]
```diff
@@ -221,6 +229,73 @@ static void addSanitizerCoveragePass(const PassManagerBuilder &Builder,
...
+static void addSanitizers(PassBuilder &PB) {
+
+  auto addXSanPasses = [&](SanitizerBits mask, void (*passAdder)(const PassManagerBuilder &Builder,
+                                                           PassManagerBase &PM)) {
+    if (opts::isSanitizerEnabled(mask) {
+      builder.addExtension(PassManagerBuilder::EP_OptimizerLast, passAdder);
+      builder.addExtension(PassManagerBuilder::EP_EnabledOnOptLevel0, passAdder);
+    }
+  }
+  addXSanPasses(opts::AddressSanitizer,  addAddressSanitizerPasses);
```
Comment (JohanEngelen) [inline gen/optimizer.cpp:243]: did you check whether this puts the passes in the same order as with the old passmanager?

**Comment by JohanEngelen** [comment]
I think it is better to make new/old passmanager selectable through a runtime option (commandflag) in the transition period. Easier to investigate bugs that way.

**Comment by thewilsonator** [comment]
superseded by https://github.com/ldc-developers/ldc/pull/4105

**Pull request closed without merging**